dags/process_elife_zip_dag.py

```diff
@@ -53,14 +53,28 @@
 }
 
 
-def get_expected_elife_article_name(file_name: str) -> str:
-    pattern = r'^\w+-\w+'
-    article_name = re.search(pattern, file_name)
-    error_message = ('%s is malformed. Expected archive name to start with '
-                     'any number/character, hyphen, any number/character (%s)'
-                     'example: name-id.extension' % (file_name, pattern))
-    assert article_name and file_name.startswith(article_name.group()), error_message
-    return article_name.group() + '.xml'
+def get_article_file_name(zip_file_name: str) -> str:
+    article_file = None
+
+    with TemporaryFile(dir=TEMP_DIRECTORY) as temp_zip_file:
+        s3 = get_s3_client()
+        s3.download_fileobj(
+            Bucket=SOURCE_BUCKET,
+            Key=zip_file_name,
+            Fileobj=temp_zip_file
+        )
+
+        for zipped_file in ZipFile(temp_zip_file).namelist():
+            if zipped_file.endswith('.xml'):
+                xml = etree.parse(BytesIO(ZipFile(temp_zip_file).read(zipped_file)))
```

Comment:
should create `ZipFile(temp_zip_file)` in a single place?

Reply:
updated.